src/main/java/minicraft/level/tile/farming/FarmTile.java

```diff
@@ -44,18 +49,30 @@ public boolean interact(Level level, int xt, int yt, Player player, Item item, D
         return false;
     }
 
-    @Override
-    public boolean tick(Level level, int xt, int yt) {
-        int age = level.getData(xt, yt);
-        if (age < 5) level.setData(xt, yt, age + 1);
-        return true;
-    }
+	@Override
+	public boolean tick(Level level, int xt, int yt) {
+		int moisture = level.getData(xt, yt) & 0b111;
+		if (Arrays.stream(level.getAreaTiles(xt, yt, 4)).anyMatch(t -> t instanceof WaterTile)) { // Contains water.
+			if (moisture < 7 && random.nextInt(10) == 0) { // hydrating
+				level.setData(xt, yt, moisture + 1);
+				return true;
+			}
+		} else if (moisture > 0 && random.nextInt(10) == 0) { // drying
+			level.setData(xt, yt, moisture - 1);
+			return true;
+		} else if (moisture == 0 && random.nextInt(10) == 0) {
+			level.setTile(xt, yt, Tiles.get("dirt"));
+			return true;
+		}
```

Comment:
If you are talking about the mechanic about turning back into a dirt tile, it might not be a good idea to turn it into a dirt tile if it is a crop as they will be lost.

Reply:
It would make more sense. Plus you wouldn't be breaking the DRY principle. If walking on a crop returns it to dirt, then drying a crop should return it to dirt. Use inheritance.

Reply:
There is a player near them when walking on them, but there is most probably not a player near them when dehydrating.

Reply:
Okay, this is fine.